app/context.go

```diff
@@ -0,0 +1,100 @@
+package app
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+)
+
+type ContextKey string
+
+const (
+	TraceIDKey  ContextKey = "trace_id"
+	UserIDKey   ContextKey = "user_id"
+	TenantIDKey ContextKey = "tenant_id"
+)
+
+// Context carries the context of the current execution.
+type Context struct {
+	// original context
+	context.Context
+}
+
+// UserID returns the user id
+func (sc *Context) UserID() (string, bool) {
+	userID := sc.Context.Value(UserIDKey)
+
+	if id, ok := userID.(string); ok {
+		return id, ok
+	}
+
+	return "", false
+}
+
+// SetUserID sets the user id
+func (sc *Context) SetUserID(userID string) *Context {
+	sc.Context = context.WithValue(sc.Context, UserIDKey, userID)
+
+	return sc
+}
+
+// TenantID returns the tenant id
+func (sc *Context) TenantID() (string, bool) {
+	tenantIDKey := sc.Context.Value(TenantIDKey)
+
+	if id, ok := tenantIDKey.(string); ok {
+		return id, true
+	}
+
+	return "", false
+}
+
+// SetTenantID sets the user id
+func (sc *Context) SetTenantID(tenantID string) *Context {
+	sc.Context = context.WithValue(sc.Context, TenantIDKey, tenantID)
+
+	return sc
+}
+
+// SetTraceID sets the trace id
+func (sc *Context) SetTraceID(traceID string) *Context {
+	sc.Context = context.WithValue(sc.Context, TraceIDKey, traceID)
+
+	return sc
+}
+
+// TraceID returns the trace identifier for the current flow
+func (sc *Context) TraceID() string {
+	traceID := sc.Context.Value(TraceIDKey)
+
+	if id, ok := traceID.(string); ok {
+		return id
+	}
+
+	return ""
+}
+
+// FromContext returns a new Context from a context.Context
+func FromContext(ctx context.Context) Context {
+	appCtx := NewContext(ctx)
+
+	if traceID, ok := ctx.Value(TraceIDKey).(string); ok {
```

Comment:
I don't think these `3 if` block is needed. I mean the context has the value, and calling the methods will simply set something that is already there

Reply:
Yes, this was my brain fart 🐛